x/ibc/03-connection/keeper/handshake_test.go

```diff
@@ -289,6 +293,111 @@ func (suite *KeeperTestSuite) TestConnOpenConfirm() {
 	}
 }
 
+// TestOpenTryVersionNegotiation uses tests the result of the version negotiation done during
+// the handshake procedure OpenTry.
+// TODO: following the refactor of this file based on ibc testing package, these cases
+// should become individual cases for the OpenTry handshake.
+// https://github.com/cosmos/cosmos-sdk/issues/5558
+func TestOpenTryVersionNegotiation(t *testing.T) {
+	testCases := []struct {
+		name                 string
+		counterpartyVersions []string
+		expPass              bool
+	}{
+		{"valid counterparty versions", types.GetCompatibleVersions(), true},
+		{"empty counterparty versions", []string{}, false},
+		{"no counterparty match", []string{"(version won't match,[])"}, false},
```

Comment:
Is this really a valid serialisation (`"(version won't match,[])"`)?

Reply:
yup